subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/gcc/GccLinker.java

```diff
@@ -41,7 +41,7 @@
     @Override
     protected Action<BuildOperationQueue<CommandLineToolInvocation>> newInvocationAction(final LinkerSpec spec, List<String> args) {
         final CommandLineToolInvocation invocation = newInvocation(
-            "linking " + spec.getOutputFile().getName(), spec.getOutputFile().getParentFile(), args, spec.getOperationLogger());
+            "linking " + spec.getOutputFile().getName(), args, spec.getOperationLogger());
```

Comment:
could we explicitly use the projectDir (and keep the same signature) for now?

Reply:
How do you think we can get the projectDir at this stage? We are pretty far down the chain. We could pass it via the `LinkerSpec`, however, it feels more like a hack than anything else. I propose to leave it like this and clean up this once we get around to improving the compiler infrastructure. What do you think?

Reply:
SGTM